<!-- -*- mode: XML -*- -->

<launch>

  <node pkg="lio" type="lio_processor_node" name="lio_processor" output="log">
  	<param name="rad_diff" type="double" value="1.0" />
  	<param name="infer_start_ori" type="bool" value="false" />
  </node>
  <node pkg="lio" type="lio_estimator_node" name="lio_estimator" output="screen">
    <param name="config_file" type="string" value="$(find lio)/config/my_test_16_indoor.yaml" />
  </node>

  <!-- <node pkg="lio" type="lio_save_pose_to_file" name="lio_save_odom_pose" output="screen">
    <param name="output_local_odom" type="string" value="/home/belight/Dataset/LIO/test20200114/local_odom_test1.txt" />
    <param name="output_map_to_init" type="string" value="/home/belight/Dataset/LIO/test20200114/map_to_init_test1.txt" />
  </node> -->

  <node pkg="lio" type="lio_save_local_mapped_pose" name="lio_save_pose" output="screen">
    <param name="output_local_mapped_odom" type="string" value="/home/belight/Dataset/LIO/test20200114/test1/local_mapped_odom7.txt" />
  </node>

  <arg name="rviz" default="true" />
  <arg name="enable_4d" default="true" />

  <node pkg="lio" type="lio_map_builder_node" name="lio_map_builder" output="screen">
  	<param name="compact_data" type="bool" value="false" />
  	<param name="enable_4d" type="bool" value="$(arg enable_4d)" />
  	<remap from="/full_odom_cloud" to="/local/full_points"/>
  	<remap from="/laser_cloud_surf_last" to="/local/surf_points"/>
  	<remap from="/laser_cloud_corner_last" to="/local/corner_points"/>
  	<remap from="/laser_odom_to_init" to="/local_laser_odom"/>
  </node> 

  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find lio)/rviz_cfg/my_test_16.rviz" />
  </group>  

</launch>